pkg/cluster/reconcile.go

```diff
@@ -45,7 +45,7 @@ func (c *Cluster) reconcileSize() error {
 		// Remove extra pods as required in order to meet the desired size.
 		// As we remove each pod, we must update the config secret so that routes are re-computed.
 		for idx := currentSize - 1; idx >= desiredSize; idx-- {
-			if err := c.removePod(pods[idx]); err != nil {
+			if err := c.tryGracefulPodDeletion(pods[idx]); err != nil {
```

Comment:
maybe later we could look into being able to hook the operator so that on `kubectl delete natscluster` it calls deletion of the pods gracefully too

Reply:
@wallyqs can you please open a feature request issue describing the requirements?